neighbour-server/src/main/java/no/vegvesen/ixn/federation/server/NeighbourRestController.java

```diff
@@ -50,12 +56,14 @@ public NeighbourRestController(NeighbourService neighbourService,
 	public SubscriptionResponseApi requestSubscriptions(@RequestBody SubscriptionRequestApi neighbourSubscriptionRequest) {
 		NeighbourMDCUtil.setLogVariables(properties.getName(), neighbourSubscriptionRequest.getName());
 		logger.debug("Received incoming subscription request: {}", neighbourSubscriptionRequest.toString());
-
-
 		// Check if CN of certificate matches name in api object. Reject if they do not match.
 		certService.checkIfCommonNameMatchesNameInApiObject(neighbourSubscriptionRequest.getName());
 		logger.debug("Common name of certificate matched name in API object.");
 
+		if(neighbourSubscriptionRequest.getSubscriptions() == null || neighbourSubscriptionRequest.getSubscriptions().isEmpty()){
```

Comment:
Add this in NeighbourService instead. Doing double work here. 
Also do a check to see if selector is valid.